### PR TITLE
Add mamba support and update pangolin lineage determination

### LINKED ISSUE #56: Mamba support
Software dependencies are currently managed by conda. Managing these dependencies with Mamba could be a significant speed improvement

### LINKED ISSUE #60: Update pangolin version in pipeline
We use a relatively old version of pangolin. We should update to version 4.3.

### PR DESCRIPTION
**Pull request by johausmann**
Closes #56 
Closes #60

### CODE REVIEW

**Comment by johausmann** [comment]
I updated the giltab and github CI files to use mamba instead of conda

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -8,11 +8,23 @@ before_script:
...
-  - rm -f Miniconda3-latest-Linux-x86_64.sh
-  - conda --version
+  - bash Mambaforge-Linux-x86_64.sh -b -p /usr/local/bin/conda
+  - rm -f Mambaforge-Linux-x86_64.sh
+  - source /usr/local/bin/conda/etc/profile.d/conda.sh
+  - source /usr/local/bin/conda/etc/profile.d/mamba.sh
+  - which mamba 
+  - mamba --version
+  # this is a workaround for https://github.com/mamba-org/mamba/issues/1993
+  - echo "use_lockfiles:" >> ~/.condarc
```
Comment (priesgo) [inline .gitlab-ci.yml:20]: this is an issue we had in LOHHLA, the file has to be `~/.mambarc`

**Review by priesgo — APPROVED** [review]
Approving so you can merge, just one issue in the CI setup, see comment.

File: .gitlab-ci.yml (hunk 1 above)
Comment (johausmann) [reply .gitlab-ci.yml:20]: Thanks. I'll change it before merging

**Pull request merged**